executor/split.go

```diff
@@ -172,19 +172,16 @@ func (e *SplitIndexRegionExec) getSplitIdxPhysicalKeysFromValueList(physicalID i
 }
 
 func (e *SplitIndexRegionExec) getSplitIdxPhysicalStartAndOtherIdxKeys(physicalID int64, keys [][]byte) [][]byte {
-	// Split in the start of the index key.
-	startIdxKey := tablecodec.EncodeTableIndexPrefix(physicalID, e.indexInfo.ID)
-	keys = append(keys, startIdxKey)
-
-	// Split in the end for the other index key.
-	for _, idx := range e.tableInfo.Indices {
-		if idx.ID <= e.indexInfo.ID {
-			continue
-		}
-		endIdxKey := tablecodec.EncodeTableIndexPrefix(physicalID, idx.ID)
-		keys = append(keys, endIdxKey)
-		break
+	// 1. Split in the start key for the index if the index is not the first index.
+	// For the first index, split the start key is useless.
+	if len(e.tableInfo.Indices) > 0 && e.tableInfo.Indices[0].ID != e.indexInfo.ID {
+		startKey := tablecodec.EncodeTableIndexPrefix(physicalID, e.indexInfo.ID)
+		keys = append(keys, startKey)
 	}
+
+	// 2. Split in the end key.
+	endKey := tablecodec.EncodeTableIndexPrefix(physicalID, e.indexInfo.ID+1)
```

Comment:
After this PR, it seems we must have `endKey`, which is different from the old code. Why change it, and do we have a test for this code.

Reply:
suppose the table is:  `create table t (a int, index idx(a))`;
then execute `split table t index idx between (0) and (20000) regions 2;` 

before this PR, the table region will be:

```sql
region1:  [tid_,          tid_i_1_)  
region2:  [tid_i_1_,      tid_i_1_10000)   
region3:  [tid_i_1_10000, nextTid_)  
```

As you can see,  the index data that `a > 10000` and the record data will write into region3, this is not expected, the record data should write into a separate region.

Reply:
https://github.com/pingcap/tidb/pull/17070/files#diff-6d9a596d0d155b4342a598ba359efc8bR4494 is the test.

Reply:
Already has unit test for this.

Reply:
Got it.